rest/doc_api.go

```diff
@@ -370,14 +370,25 @@ func (h *handler) handlePutDocReplicator2(docid string, roundTrip bool) (err err
 	if err != nil {
 		return err
 	}
-	if bodyBytes == nil || len(bodyBytes) == 0 {
-		return base.ErrEmptyDocument
-	}
 
-	newDoc := &db.Document{
-		ID: docid,
+	hasAttachments := bytes.Contains(bodyBytes, []byte(db.BodyAttachments))
+	hasExpiry := bytes.Contains(bodyBytes, []byte(db.BodyExpiry))
+
+	var body db.Body
+	var newDoc *db.IncomingDocument
+	newDoc = db.NewIncomingDocument(bodyBytes)
```

Comment:
same thing - newDoc is not usable until it has doc ID and rev ID on it. We should try not to initialize it until we are able to do that.

Reply:
Will go back to this once handleRev is done....